src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameGapCheck.java

```diff
@@ -0,0 +1,218 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.direction.EdgeDirectionComparator;
+
+/**
+ * This check identifies Edges with no name Tag that are between two other Edges with the same name
+ * Tag, OR the Edge has a name Tag but does not equal the name Tag of the Edges that it is between.
+ *
+ * @author sugandhimaheshwaram
+ */
+public class RoadNameGapCheck extends BaseCheck
+{
+    /**
+     * This checks if the connected edges have same direction or not.
+     */
+    class EdgePredicate
```

Comment:
Nice use of EdgeDirectionComparator. Why do we wrap EdgeDirectionComparator in EdgePredicate? Seems like we could just use the former directly

Reply:
I use EdgePredicate class for reusability purpose. I was using filter in flag() method in two places one for inEdge and one for outEdge. Do not like to repeat the code. Cannot add a method outside it because I need two parameters for comparison.